lms/services/assignment.py

```diff
@@ -74,6 +82,7 @@ def update_assignment(self, request, assignment, document_url, group_set_id):
         assignment.is_gradable = self._misc_plugin.is_assignment_gradable(
             request.lti_params
         )
+        assignment.course_id = course.id
```

Comment:
We want to update this all the time, we always trust the info from the LMS.

If we need any insights of the details of ways an assignment jumped between courses in some edge cases we can always refer to assigment_grouping for more details.